lib/gceimgutils/gceremoveimg.py

```diff
@@ -0,0 +1,177 @@
+        """Find the images to remove"""
+        owned_images = utils.get_project_images(
+            self.project, self.credentials, True
+        )
+        if self.image_name:
+            return utils.find_images_by_name(
+                owned_images,
+                self.image_name,
+                self.log
+            )
+        elif self.image_name_fragment:
+            return utils.find_images_by_name_fragment(
+                owned_images,
+                self.image_name_fragment,
+                self.log
+            )
+        elif self.image_name_match:
+            try:
+                return utils.find_images_by_name_regex_match(
+                    owned_images,
+                    self.image_name_match,
+                    self.log
+                )
+            except Exception:
+                msg = 'Unable to complie regular expression "%s"'
+                msg = msg % self.image_name_match
+                raise GCERemoveImgException(msg)
+        else:
+            msg = 'No remove image condition set. Should not reach '
+            msg += 'this point.'
+            raise GCERemoveImgException(msg)
+
+    # ---------------------------------------------------------------------
+    def print_remove_info(self):
+        """Print information about images that would be deleted"""
+        images = self._get_images_to_remove()
+        images_ok = self._check_images_boundary_condition(images)
+
+        if not images_ok:
+            # While the boundary conditions are not met, we are just printing
+            # information thus this is not considered a failure
+            return True
+
+        header_msg = 'Would remove image'
+        if len(images) > 1:
+            header_msg += 's'
+
+        self.log.info(header_msg)
+        for image in images:
+            self.log.info('\t\t%s' % image.get('name'))
+
+        return True
+
+    # ---------------------------------------------------------------------
+    def remove_images(self):
+        """Remove the images"""
+        api = self._get_api()
+        images = self._get_images_to_remove()
+        images_ok = self._check_images_boundary_condition(images)
+
+        if not images_ok:
+            raise GCERemoveImgException('Image ambiguity')
+
+        for image in images:
+            delete = 'True'
+            if self.confirm:
+                delete = self._query_yes_no(image)
+
+            if delete:
+                image_name = image.get('name')
+                try:
+                    api.images().delete(
+                        project=self.project, image=image_name
+                    ).execute()
+                except HttpError:
+                    self.log('Unable to remove image: "%s"' % image_name)
+            else:
+                continue
+
+    # ---------------------------------------------------------------------
+    def _query_yes_no(self, image):
+        yes = {'yes', 'y', 'ye', ''}
+        no = {'no', 'n'}
```

Comment:
would it be safer to not delete image by default ?
As it is, one is deleting images and reading the names while deleting, delete the wrong/undesired image 
could happen.

Thoughts ?